### PR TITLE
Create a new migration for cpython3.10

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xhochy** [comment]
We shouldn't merge this until https://github.com/conda/conda/pull/10970 is relased?

File: recipe/migrations/python310.yaml
Comment (mariusvniekerk) [reply recipe/migrations/python310.yaml:28]: The dependency resolution will block in any case if you need a numpy and we haven't built it yet.  Since this is a zip_keys set we HAVE to have a value in here, so may as well pick the same one we are using for the rest of them.

There are a bunch of packages that we'll build before we get to a numpy in any case.  

If we need to bump the numpy version in case we cannot build 1.19 for 3.10 at all we can just release a new migrator version.

Comment (beckermr) [reply recipe/migrations/python310.yaml:28]: Ohhhh. Yes, we'd need to nackport that build or add an ABI branch.

**Review by isuruf — DISMISSED** [review]
We need to figure out aarch64 builds before merging this.

**Comment by jakirkham** [comment]
Not seeing that covered in the last meeting's notes. Could you please clarify here? A one sentence summary should be sufficient :)

**Comment by isuruf** [comment]
Quoting from the notes.

```
(IF) aarch64 builds
    * cloud.drone.io is slow
    * We have $3000 credits from Oracle for a year
        * Enough for 8 runners with 4 vCPUs and 24 GB for a
          year
    * options
        1. host our own drone server/runners
        2. host our own azure agents
        3. cross compile
            - test using qemu
            - pypy builds
        4. emulate
 ```

File: recipe/migrations/python310.yaml
Comment (h-vetinari) [reply recipe/migrations/python310.yaml:28]: > @mariusvniekerk: If we need to bump the numpy version in case we cannot build 1.19 for 3.10 at all we can just release a new migrator version.

Numpy will only support 3.10 [earliest](https://github.com/numpy/numpy/pull/20098) from 1.21.3, so backporting to 1.19 sounds tricky.

**Comment by mariusvniekerk** [comment]
Running a simplish check shows that this will migrate about 2529 packages which is similar to what we did last time round.

File: recipe/migrations/python310.yaml
Comment (rgommers) [reply recipe/migrations/python310.yaml]: For the record, +1 to using `1.21`. Note that `1.21.3` is not yet out (that's the first version that will have wheels for 3.10) but `1.21.2` will build just fine with Python 3.10.

There's no point backporting fixes to `1.20.x` or earlier, that's just extra effort and means things get out of sync with the officially supported releases and with PyPI.

File: recipe/migrations/python310.yaml
Comment (h-vetinari) [inline recipe/migrations/python310.yaml:14]:
```suggestion
            - 3.10.* *_cpython  # new entry
            - 3.6.* *_73_pypy
            - 3.7.* *_73_pypy

```

**Comment by beckermr** [comment]
See the discussion on this issue, namely the aarch64 and ppc64le builds.

**Comment by isuruf** [comment]
Well, what would you rather like?
1. Start right now
   a. Have the migration go for months
   b. Disrupt everybody else working on conda-forge by having drone being queued for hours/days
   c. maintainers/core being asked constantly to restart migrations because of travis being down

2. Do one of the other alternatives which will make the migration go smoothly

**Comment by beckermr** [comment]
I agree starting now would be much too disruptive. 

To be more specific on timelines, we are exploring the options above. Each of them will take 1-2 weeks on their own to put into place due to various bugs, migrations, and/or needed features. After it actually starts, I expect it to last at least several weeks if not more (especially if we use emulated builds in places). So I would not expect a working python 3.10 stack w/ most dependencies that install on 3.9 from conda-forge for at least 1-2 months. 

Given that timescale, I think we have maybe a week or two more to decide what to do before we are really delaying things in a large way.

The good news is that we have decent default/backup plans that are largely workable, so no matter what we have not yet hit a point where we simply cannot build conda-forge anymore due to a CI limit. 

I know this is frustrating, but we'll get there!

**Comment by h-vetinari** [comment]
BTW, I'd volunteer to nurse the migrator PRs, assuming there's a way I can be (temporarily?) granted the requisite rights.

**Comment by isuruf** [comment]
Since drone.io seems to be doing okay today, we can try with 5 PR limit and increase the limit later. If there's no other objections, I'll merge tonight.

**Review by ocefpaf — APPROVED** [review]

**Comment by ocefpaf** [comment]
> If there's no other objections, I'll merge tonight.

**Pull request merged**
